game-core/src/main/java/games/strategy/engine/data/unit/ability/CombatUnitAbility.java

```diff
@@ -0,0 +1,106 @@
+package games.strategy.engine.data.unit.ability;
+
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.data.UnitType;
+import games.strategy.triplea.delegate.battle.BattleState;
+import java.util.Collection;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.AccessLevel;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NonNull;
+import lombok.experimental.FieldDefaults;
+import lombok.experimental.NonFinal;
+
+/** Unit Ability for combat situations */
+@Data
+@FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE)
+@Builder(toBuilder = true)
+public class CombatUnitAbility {
+
+  enum DiceType {
+    // Use AA dice (attackAA, offensiveAttackAA)
+    AA,
+    // Use bombard dice (bombard or attack if bombard is empty)
+    BOMBARD,
+    // Use normal dice (attack, defense)
+    NORMAL
+  }
+
+  /** The name of this unit ability that will be shown in the Battle UI */
+  @NonNull String name;
+
+  /** The unit types that have this ability */
+  @NonFinal Collection<UnitType> attachedUnitTypes;
+
+  /** The unit types that can be targeted */
+  @NonNull Collection<UnitType> targets;
+
+  /** The type of dice that will be rolled. */
+  @Builder.Default DiceType diceType = DiceType.NORMAL;
+
+  /** The side(s) that the unit needs to be on to use the ability */
+  @Builder.Default
+  Collection<BattleState.Side> sides = List.of(BattleState.Side.OFFENSE, BattleState.Side.DEFENSE);
+
+  /** How many combat rounds this ability can be used. -1 means unlimited */
+  @Builder.Default int round = -1;
+
+  /**
+   * Can the casualties return fire after the battle phase that this UnitAbility is active?
+   *
+   * <p>Casualties that are created in the battle phase where this UnitAbility is active can always
+   * return fire since they are firing simultaneously
+   */
+  @Builder.Default boolean returnFire = true;
+
+  /** Will these units commit suicide when they have a hit? */
+  @Builder.Default boolean suicideOnHit = false;
```

Comment:
Some units are suicide after attack, is that another attribute that would be added, or is it related to this attribute?

Reply:
Could you send me a link to refresh my memory? I've my hand in many TripleA pots, lots going on.

I think I'm just confused as those seem like combat abilities, `isSuicideOnAttack` & `isSuicideOnDefense`, and I'm not sure how that is different then from `isSuicideOnHit`, and why only the latter is represented here but no the other two. Can you suggest a way to make this more clear to me and other future maintainers reading through this code?

Reply:
`isSuicideOnAttack` and `isSuicideOnDefense` means that the unit will have one chance to fire and then will commit suicide whether it hit or not.  `isSuicideOnHit` means that the unit will fire every round and it only commits suicide when it has a successful hit.

Now that you mention it, `isSuicideOnAttack` and `isSuicideOnDefense` should probably be represented as a unit ability.  I`m thinking `Collection<Side> commitSuicide`.  And maybe I should chance `suicideOnHit` to `Collection<Side> commitSuicideAfterSuccessfulHit`.

I also just checked and I can't find a map that is using `isSuicideOnHit` so I'm not sure how it would be used.

Reply:
Done in https://github.com/triplea-game/triplea/pull/8646/commits/6b79c49cd64666026b410bb54293f8daa3410537

Also, since `isSuicideOnHit` isn't used in any map, I'm going to change its functionality a little bit.  Currently, the engine splits up all of the `isSuicideOnHit` unit types so that when a hit occurs, it can just kill one of the units since it knows that all of the units are the same unit types.  But I'm going to change it so that`isSuicideOnHit` unit types can be combined and it will just randomly pick out a unit to suicide.  For non-low luck, it will handle the case where a unit type might have different dice sides.  For low luck, it will just randomly select units.

I think the functionality is similar enough that it should be ok.